esp/platform/application_config.cpp

```diff
@@ -89,18 +89,29 @@ static void copyDirectories(IPropertyTree *target, IPropertyTree *src)
     }
 }
 
-void addAuthDomains(const char *authType, IPropertyTree *legacyEsp)
+void addAuthDomains(IPropertyTree *appEsp, IPropertyTree *legacyEsp)
 {
-    if (isEmptyString(authType)) //backward compatible
-        authType = "AuthPerRequestOnly";
+    IPropertyTree *authDomainsTree = appEsp->queryPropTree("authDomains");
+    if (authDomainsTree)
```

Comment:
@wangkx if you agree, we can get rid of the "authDomains" tag here and skip to accessing one authDomain below.  Changing the Yaml input but keeping the output legacy xml the same.